libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ObjectDetectionModelOverview.tsx

```diff
@@ -77,11 +87,80 @@ export class ObjectDetectionWidgets extends React.PureComponent<IObjectDetection
             localization.ModelAssessment.ModelOverview.iouthresholdDropdown
           }
           max={100}
+          defaultValue={70}
           className={this.props.classNames.slider}
+          onChanged={this.onIoUThresholdChange}
           valueFormat={(value: number): string => `IoU=${value}%`}
           showValue
         />
       </Stack.Item>
     );
   }
+
+  private onAggregateMethodChange = (
+    _: React.FormEvent<IComboBox>,
+    item?: IComboBoxOption
+  ): void => {
+    console.log(item);
+    if (item) {
+      console.log("entered aggregate method change");
+
+      // State directly changed instead of setState method
+      // to avoid memory leaks with unmounted Model Overview.
+      this.props.modelOverview.state.aggregateMethod = item.text.toString();
+
+      if (this.props.modelOverview.state.datasetCohortChartIsVisible) {
+        this.props.modelOverview.updateDatasetCohortStats();
+      } else {
+        this.props.modelOverview.updateFeatureCohortStats();
+      }
+
+      this.props.modelOverview.logButtonClick(
+        TelemetryEventName.ModelOverviewMetricsSelectionUpdated
+      );
+    }
+  };
+
+  private onClassNameChange = (
+    _: React.FormEvent<IComboBox>,
+    item?: IComboBoxOption
+  ): void => {
+    console.log(item);
+    if (item) {
+      console.log("entered class name change");
+
+      // State directly changed instead of setState method
+      // to avoid memory leaks with unmounted Mount Overview.
+      this.props.modelOverview.state.className = item.text.toString();
```

Comment:
we should never modify state directly like this - instead you can use setState method